library/std/src/thread/local.rs

```diff
@@ -1028,10 +1025,16 @@ pub mod fast {
         // `Option<T>` to `None`, and `dtor_state` to `RunningOrHasRun`. This
         // causes future calls to `get` to run `try_initialize_drop` again,
         // which will now fail, and return `None`.
-        unsafe {
+        //
+        // Wrap the call in a catch to ensure unwinding is caught in the event
+        // a panic takes place in a destructor.
+        if let Err(_) = panic::catch_unwind(panic::AssertUnwindSafe(|| unsafe {
+            let ptr = ptr as *mut Key<T>;
             let value = (*ptr).inner.take();
             (*ptr).dtor_state.set(DtorState::RunningOrHasRun);
             drop(value);
+        })) {
+            rtabort!("destructor panicked");
```

Comment:
Can you change the panic message to "thread local panicked on drop"? Then it matches the message of thread results:

https://github.com/rust-lang/rust/blob/6a20f7df5755d8c6b68110d2d0391a7b03268e77/library/std/src/thread/mod.rs#L1424-L1428

Reply:
Great hint, thank & done